packages/go/stbernard/graph/transform.go

```diff
@@ -0,0 +1,346 @@
+		return fmt.Errorf("error getting ingest file paths from test directory %v", err)
+	} else if err = ingestData(ctx, ingestFilePaths, database); err != nil {
+		return fmt.Errorf("error ingesting data %v", err)
+	} else if nodes, edges, err := getNodesAndEdges(ctx, database); err != nil {
+		return fmt.Errorf("error retrieving nodes and edges from database %w", err)
+	} else if arrows, err := transformToArrows(nodes, edges); err != nil {
+		return fmt.Errorf("error transforming nodes and edges to arrows format %w", err)
+	} else if jsonBytes, err := json.MarshalIndent(arrows, "", "  "); err != nil {
+		return fmt.Errorf("error occurred while marshalling arrows into bytes %w", err)
+	} else if err := os.MkdirAll(filepath.Dir(outPath), 0o755); err != nil {
+		return fmt.Errorf("creating output directory: %w", err)
+	} else {
+		return os.WriteFile(outPath, jsonBytes, 0o644)
+	}
+}
+
+type Position struct {
+	X int `json:"x"`
+	Y int `json:"y"`
+}
+
+type Node struct {
+	ID         string            `json:"id"`
+	Position   Position          `json:"position"`
+	Caption    string            `json:"caption"` // name of node -- if not, use object_id
+	Label      []string          `json:"label"`   // kinds
+	Properties map[string]string `json:"properties"`
+}
+
+type Relationship struct {
+	ID         string            `json:"id"`
+	Label      string            `json:"label"` // kind
+	From       string            `json:"fromId"`
+	To         string            `json:"toId"`
+	Properties map[string]string `json:"properties"`
+}
+
+type Arrows struct {
+	Nodes         []Node         `json:"nodes"`
+	Relationships []Relationship `json:"relationships"`
+}
+
+func ingestData(ctx context.Context, filepaths []string, database graph.Database) error {
+	var errs []error
+
+	schema, err := upload.LoadIngestSchema()
+	if err != nil {
+		return fmt.Errorf("error loading ingest schema %v", err)
+	}
+
+	for _, filepath := range filepaths {
+		err := database.BatchOperation(ctx, func(batch graph.Batch) error {
+			timestampedBatch := graphify.NewTimestampedBatch(batch, time.Now().UTC())
+
+			readOpts := graphify.ReadOptions{IngestSchema: schema, FileType: model.FileTypeJson, ADCSEnabled: true}
+
+			file, err := os.Open(filepath)
+			if err != nil {
+				return fmt.Errorf("error opening JSON file %s: %v", filepath, err)
+			}
+			defer file.Close()
+
+			// ingest file into database
+			err = graphify.ReadFileForIngest(timestampedBatch, file, readOpts)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("error ingesting file %s: %w", filepath, err))
+			}
+
+			return nil
+		})
+		if err != nil {
+			return fmt.Errorf("unrecoverable error occurred during batch operation %w", err)
+		}
+	}
+
+	if len(errs) > 0 {
+		var errStrings []string
+		for _, err := range errs {
+			errStrings = append(errStrings, err.Error())
+		}
+		slog.Warn("errors occurred while ingesting files", "errors", errStrings)
+	}
```

Comment:
We could just `return errors.Join(errs...)` and log the error further up. I'd consider any errors to be `error` level in this context, since missing ingest files is going to make the final result unusable for intended purposes

Reply:
If we really want them to be a warn for some reason, we could `return fmt.Errorf("%w: %w", ErrIngestWarning, errors.Join(errs...))` to give a sentinel for the caller to distinguish between generic errors and this warning case (requires defining a sentinel called ErrIngestWarning)